src/store/modules/rate/index.js

```diff
@@ -41,22 +41,26 @@ const actions = {
         })
     })
   },
-  getHistoryRates ({ commit }, { date, coin }) {
+  getHistoryRates ({ state, commit }, { date, coin }) {
     const url = getEnpointUrl('infoservice')
     const key = date
-    date = coin === 'ADM' ? (Math.floor((date * 1000 + EPOCH) / 1000)) : Math.floor(date / 1000)
-    return new Promise((resolve, reject) => {
-      axios
-        .get(`${url}/getHistory?timestamp=${date}`)
-        .then((res) => {
-          const rates = res.data.result[0].tickers
-          commit('setHistoryRates', { name: key, value: rates })
-          resolve(res)
-        })
-        .catch((err) => {
-          reject(err)
-        })
-    })
+    if (state.historyRates[key]) {
+      return state.historyRates[key]
+    } else {
+      date = coin === 'ADM' ? (Math.floor((date * 1000 + EPOCH) / 1000)) : Math.floor(date / 1000)
```

Comment:
You can use `adamant.epochTime`

Reply:
Better make this change before function call.
I. e., getHistoryRates(timestamp), where timestamp is already in correct format.